articles/stock-price-prediction-using-python/index.md

```diff
@@ -0,0 +1,170 @@
+```python
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+from sklearn.preprocessing import MinMaxScaler
+from keras.models import Sequential, load_model
+from keras.layers import LSTM, Dense, Dropout
+```
+We are going to use `numpy` for scientific operations, `pandas` to modify our dataset, `matplotlib` to visualize the results, `sklearn` to scale our data, and `keras` works as a wrapper to low-level libraries like TensorFlow or Theano high-level neural networks library.
+
+### Let's build our app!
+First of all, if you take a look at the dataset you need to know that "open" column represents the opening price for the stock at that "date" column, and the "close" column is the closing price on that day. "High" column represents the highest price reached that day, and "Low" column represents the lowest price.
+
+#### Loading data
+Let's upload our dataset(please ignore the following code if you are not using Google Colab):
+```python
+from google.colab import files
+dataset = files.upload()
+```
+Then press "choose files" and upload the dataset.
+
+#### Feature Extraction
+After that let's get the number of the trading days by:
+```python
+df.shape
+```
+The result will be `(2392, 7)`.
+
+To make it as simple as possible we will just use one variable which is the "open" price.
+```python
+df = df['Open'].values
+df = df.reshape(-1, 1)
+```
+The `reshape` allows you to add dimensions or change the number of elements in each dimension. We are using `reshape(-1, 1)` because we have just one dimension in our array, so `numby` will create the same number of our rows and add one more axis which is 1 to be the second dimension.
+
+Now let's split the data into training and testing sets:
+```python
+dataset_train = np.array(df[:int(df.shape[0]*0.8)])
+dataset_test = np.array(df[int(df.shape[0]*0.8):])
+```
+We will use the `MinMaxScaler` to scale our data between zero and one. In simple words, scaling is converting the numerical data which are represented in a wide range into a smaller one.
+
+```python
+scaler = MinMaxScaler(feature_range=(0,1))
+dataset_train = scaler.fit_transform(dataset_train)
+dataset_test = scaler.transform(dataset_test)
+```
+
+Next, we will create the function which will help us to create the datasets:
+```python
+def create_dataset(df):
+    x = []
+    y = []
+    for i in range(50, df.shape[0]):
+        x.append(df[i-50:i, 0])
+        y.append(df[i, 0])
+    x = np.array(x)
+    y = np.array(y)
+    return x,y
+ ```
+Now we are going to create our training and testing data by calling our function for each one:
+```python
+x_train, y_train = create_dataset(dataset_train)
+x_test, y_test = create_dataset(dataset_test)
+ ```
+Next, we need to reshape our data to make it a 3D array in order to use it in LSTM Layer.
+```python
+x_train = np.reshape(x_train, (x_train.shape[0], x_train.shape[1], 1))
+x_test = np.reshape(x_test, (x_test.shape[0], x_test.shape[1], 1))
+```
+#### Model building

```

Comment:
Print model.summary() and include that screenshot. Explain your reasoning behind the architecture.